### PR TITLE
Document creating a custom field type

### PR DESCRIPTION
**Pull request by jorrit**
## Subject

I spent a bit of time figuring out how to create my own field type. I thought it would be helpful to document this for other users.

I am targeting this branch, because it's just documentation.

### CODE REVIEW

**Comment by jorrit** [comment]
It fails because commit https://github.com/sonata-project/dev-kit/commit/197ef06cb6ea79c6cd96c114d1735b73a2aa4a31 has yet to be merged from devkit to SonataAdminBundle.

**Comment by OskarStark** [comment]
https://github.com/sonata-project/SonataAdminBundle/pull/5945 is merged :)

File: docs/reference/field_types.rst [hunk 1]
```diff
@@ -259,3 +259,33 @@ Parameter                   Description
...
+Field types are Twig templates that are registered in the ``sonata_doctrine_orm_admin`` configuration section.
+
+.. code-block:: yaml
+
+    # config/sonata_admin.yaml
+
+    sonata_doctrine_orm_admin:
+        templates:
+            types:
+                show: # or list:
```
Comment (OskarStark) [inline docs/reference/field_types.rst]:
```suggestion
                show: # or "list"
```

File: docs/reference/field_types.rst [hunk 2]
```diff
@@ -259,3 +259,33 @@ Parameter                   Description
...
+                    dump: 'fieldtypes/show_dump.html.twig'
+
+Now add a twig file to your ``templates/`` directory. The example below uses ``@SonataAdmin/CRUD/base_show_field.html.twig`` to provide the row layout used by the show template.
+Within this base template you can override the ``field`` block to rewrite the contents of the field content cell in this row.
+
+.. code-block:: html+jinja
```
Comment (OskarStark) [inline docs/reference/field_types.rst]:
```suggestion
.. code-block:: html+twig
```

**Comment by jorrit** [comment]
Thanks for reviewing. I have updated the commit.

**Comment by OskarStark** [comment]
Thank you, I removed the changelog, its not needed for doc changes 👍

File: docs/reference/field_types.rst [hunk 3]
```diff
@@ -259,3 +259,33 @@ Parameter                   Description
             ])
         ;
     }
+
+Create your own field type
+--------------------------
+
+Field types are Twig templates that are registered in the ``sonata_doctrine_orm_admin`` configuration section.
```
Comment (phansys) [inline docs/reference/field_types.rst]: IMO, we should keep the docs as much manager agnostic as possible in this package.

Comment (VincentLanglet) [reply docs/reference/field_types.rst]: Let's just deprecate the others managers :p

Comment (jorrit) [reply docs/reference/field_types.rst]: Do you mean that I should move these additions to the docs of SonataDoctrineORMAdminBundle ?

Comment (VincentLanglet) [reply docs/reference/field_types.rst]: It should work the same for every ORM, so I think he just wants you to replace `sonata_doctrine_orm_admin` by something generic like `sonata_doctrine_xxx_admin`.

File: docs/reference/field_types.rst [hunk 4]
```diff
@@ -259,3 +259,40 @@ Parameter                   Description
...
+
+Create your own field type
+--------------------------
+
+Field types are Twig templates that are registered in the configuration section matching your ORM provider, which is one of:
+
+1.  ``sonata_doctrine_orm_admin``
+2.  ``sonata_doctrine_phpcr_admin``
+3.  ``sonata_doctrine_mongodb_admin``
+4.  ``sonata_propel_orm_admin``
```
Comment (phansys) [inline docs/reference/field_types.rst]:
```suggestion
Field types are Twig templates that are registered in the configuration section matching your model manager.
```

File: docs/reference/field_types.rst [hunk 5]
```diff
@@ -259,3 +259,40 @@ Parameter                   Description
...
+
+    # config/sonata_doctrine_orm_admin.yaml
+
+    sonata_doctrine_orm_admin:
+        templates:
+            types:
+                show: # or "list"
+                    dump: 'fieldtypes/show_dump.html.twig'
+
+Now add a twig file to your ``templates/`` directory. The example below uses ``@SonataAdmin/CRUD/base_show_field.html.twig`` to provide the row layout used by the show template.
```
Comment (phansys) [inline docs/reference/field_types.rst]:
```suggestion
Now add a twig file to your ``templates/`` directory. The example below uses ``@SonataAdmin/CRUD/base_show_field.html.twig`` to provide the row layout used by the "show" template.
```

File: docs/reference/field_types.rst [hunk 6]
```diff
@@ -259,3 +259,40 @@ Parameter                   Description
...
+Within this base template you can override the ``field`` block to rewrite the contents of the field content cell in this row.
+
+.. code-block:: html+twig
+
+    {# templates/fieldtypes/show_dump.html.twig #}
+
+    {% extends '@SonataAdmin/CRUD/base_show_field.html.twig' %}
+
+    {% block field %}
+        {{ dump(value)  }}
```
Comment (phansys) [inline docs/reference/field_types.rst]:
```suggestion
        {{ dump(value) }}
```

File: docs/reference/field_types.rst (hunk 4 above)
Comment (phansys) [reply docs/reference/field_types.rst]: Model managers should remain decoupled from this bundle. Sonata provides some common implementations (propel is deprecated), but you can even provide your own model manager.

Comment (jorrit) [reply docs/reference/field_types.rst]: How do you suggest I write that down in the documentation?

File: docs/reference/field_types.rst (hunk 6 above)
Comment (jorrit) [reply docs/reference/field_types.rst]: Fixed, thanks

File: docs/reference/field_types.rst (hunk 5 above)
Comment (jorrit) [reply docs/reference/field_types.rst]: Fixed, thanks

File: docs/reference/field_types.rst (hunk 4 above)
Comment (phansys) [reply docs/reference/field_types.rst]: The way I think it should be written is as I suggested (omitting the list of model managers currently provided by Sonata packages, keeping the example with "sonata_doctrine_orm_admin" like you already have done).

Comment (jorrit) [reply docs/reference/field_types.rst]: My apologies, I overlooked the green part of your suggestion.

**Comment by jorrit** [comment]
By the way, is there any reason why the `templates/types/show` and `templates/types/list` configuration sections are part of the model manager configuration and not of `sonata_admin`? The templates themselves are part of SonataAdmin also.

File: docs/reference/field_types.rst (hunk 4 above)
Comment (phansys) [reply docs/reference/field_types.rst]: Thank you! :+1:

**Comment by greg0ire** [comment]
Thanks @jorrit !

**Pull request merged**